### DISCUSSION TITLE
Is qdrant easy to corrupt? (like sudden restart or something like that?) Is it easy to restore after a failure?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [qdrant/qdrant#2541] [General]
Is qdrant easy to corrupt? (like sudden restart or something like that?) Is it easy to restore after a failure?

**Comment by timvisee** [comment]
That is a hard question to answer, because 'easy' is not quantifiable.

However, Qdrant is able to recover from sudden interruptions (such as a restart, a process kill or power outage). Recovery is done automatically on start, so no user intervention is required for this. Qdrant is designed with this in mind, to be able to handle sudden interruptions. So no, you cannot _easily_ corrupt the database that way.

If running in a cluster, interrupted nodes will automatically catch up with others.

There are some parameters that help with data integrity, such as `wait` and `ordering`. See [this](https://qdrant.tech/documentation/guides/distributed_deployment/#consistency-guarantees) page for information on consistency guarantees.

Note that:
- it is always recommended to make back-ups, this can be done with snapshots
- Qdrant does not have ACID guarantees, if you require that, you may have to look somewhere else
- Qdrant is better considered as vector search engine, not necessarily as vector database, which may give you a hint on what to expect
